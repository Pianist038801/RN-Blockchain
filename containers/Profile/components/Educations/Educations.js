import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import _ from 'underscore'
import Education from '../Education'
import styles from '../../../../style.js'

const Educations = (props) => {
  const educationData = props.educations

  if (!educationData) {
    return (
      <Text>Loading...</Text>
    )
  }

  if (educationData.length == 0) {
    return (
      <View>
        <Text style={styles.entityEmptyText}></Text>
      </View>
    )
  }

  const educations = []
  _.each(educationData, (education) => {
    educations.push(<Education key={education.id} education={education} />)
  })

  return (
    <View>
      {educations}
    </View>
  )
}

Educations.propTypes = {
  educations: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ])
}

export default Educations
