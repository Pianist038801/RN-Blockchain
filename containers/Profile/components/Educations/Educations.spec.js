import React from 'react'
import renderer from 'react-test-renderer'
import Educations from './index'

describe('containers/Profile/components/Educations', () => {
  describe('when prop.educations is null', () => {
    it('renders correctly', () => {
      const props = {
        educations: null
      }

      const tree = renderer.create(
        <Educations {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when prop.educations is empty', () => {
    it('renders correctly', () => {
      const props = {
        educations: []
      }

      const tree = renderer.create(
        <Educations {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when prop.educations is not empty', () => {
    it('renders correctly', () => {
      const props = {
        educations: [
          {
            id: 11,
            institution: 'Awesome Uni',
            start_date: '2016-11-04',
            end_date: '2017-01-24',
            studied: 'BSc in something',
            verified: true
          },
          {
            id: 12,
            institution: 'Westminster Uni',
            start_date: '2017-11-04',
            end_date: '2018-01-24',
            studied: 'MSc in something',
            verified: true
          }
        ]
      }

      const tree = renderer.create(
        <Educations {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
