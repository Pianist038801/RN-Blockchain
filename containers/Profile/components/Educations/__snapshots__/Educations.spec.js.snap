// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/Educations when prop.educations is empty renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#7e7e7e",
        "fontSize": 10,
      }
    }
  />
</View>
`;

exports[`containers/Profile/components/Educations when prop.educations is not empty renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderWidth": 1,
        "flex": 0.5,
        "flexDirection": "row",
        "marginBottom": 5,
        "padding": 22.5,
        "width": 600,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "resizeMode": "contain",
          "width": 75,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 22.5,
          "width": 360,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#569cee",
            "fontWeight": "bold",
          }
        }
      >
        BSc in something
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Awesome Uni
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        2016
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        - 
        2017
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderWidth": 1,
        "flex": 0.5,
        "flexDirection": "row",
        "marginBottom": 5,
        "padding": 22.5,
        "width": 600,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "resizeMode": "contain",
          "width": 75,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 22.5,
          "width": 360,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#569cee",
            "fontWeight": "bold",
          }
        }
      >
        MSc in something
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Westminster Uni
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        2017
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        - 
        2018
      </Text>
    </View>
  </View>
</View>
`;

exports[`containers/Profile/components/Educations when prop.educations is null renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Loading...
</Text>
`;
